mrmustard/physics/ansatze.py

```diff
@@ -175,6 +181,18 @@ def degree(self) -> int:
             return 0
         return self.array.shape[-1] - 1
 
+    @property
+    def polynomial_degrees(self) -> tuple[int, tuple]:
```

Comment:
Why here and not in the `DiffOpPolyExpAnsatz` class?

Reply:
This one is the new `degree` method that is defined just above in the code. 
The difference is that the degree function assumes a single variable polynomial, whereas we can have different degrees of polynomials on different wires. I just added it to the `polyexpbase` because the other one is also there. We might want to make `degree` just return the first argument of `polynomial_degrees` once the new ansatz is up and running.

Reply:
What we need for later is the number of (not) wires with polynomials on them and sometimes the degree of the polynomial on each wire. This is different from the sum of them, unless we do the decomposition - in that case the new polynomials will have the degree which is the sum of the degrees of the old ones. I also renamed this to polynomial shape.